webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +122,16 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                document.location.href = next;
```

Comment:
I don't see how you can sanitize it in Django. Pretty sure you have to do it in JS.

For the record, I'm talking about this:

https://www.owasp.org/index.php/Unvalidated_Redirects_and_Forwards_Cheat_Sheet